.changeset/khaki-ducks-give.md

```diff
@@ -0,0 +1,9 @@
+---
+'astro': patch
+---
+
+Fixes edge case on view transitions not working for some valid responses
+
+If a page was returned by the server with blank spaces before the `;` separator in the `Content-Type` header,
+the view transition would not work and a full page reload would occur.
+Now such scenarios will have the correct view transition.
```

Comment:
Proposing a slightly simplified changelog message! If anyone is interested in the nitty-gritty details, the changelog is automatically linked to this PR.

```suggestion
Fixes an edge case with view transitions where some spec-compliant `Content-Type` headers would cause a valid HTML response to be ignored.
```